### PR TITLE
Support folding subconditions into superconditions in v1 recipes

### PR DESCRIPTION
**Pull request by mgorny**
#### Description:

Support combining subconditions into superconditions while folding `if` in v1 recipes, e.g.:

```yaml
- if: A
  then: FOO
- if: A and B
  then: BAR
```

will be combined into:

```yaml
- if: A
  then:
    - FOO
    - if: B
      then: BAR
```

#### Checklist:

- [x] Pydantic model updated or no update needed

### CODE REVIEW

**Comment by mgorny** [comment]
Actually, reverting the part combining adjacent `if A and B` with `if A and C` -- my code could have accidentally slurped parts of B and C.

**Comment by beckermr** [comment]
> Actually, reverting the part combining adjacent `if A and B` with `if A and C` -- my code could have accidentally slurped parts of B and C.

Please add a test for this edge case if it is not already covered.

File: conda_forge_tick/migrators/recipe_v1.py [hunk 1]
```diff
@@ -64,6 +64,22 @@ def is_negated_condition(a: Node, b: Node) -> bool:
     return False
 
 
+def is_sub_condition(sub_node: Node, super_node: Node) -> bool:
+    return isinstance(sub_node, And) and super_node in (sub_node.left, sub_node.right)
+
+
+def get_common_condition_part(sub_cond: str, super_cond: str) -> str | None:
```
Comment (beckermr) [inline conda_forge_tick/migrators/recipe_v1.py:71]: This function will fail if "(" appears in the conditions.

**Review by beckermr — CHANGES_REQUESTED** [review]
To do this properly, we'll need to parse the expression formally I think.

File: conda_forge_tick/migrators/recipe_v1.py (hunk 1 above)
Comment (mgorny) [reply conda_forge_tick/migrators/recipe_v1.py:71]: I know. However, it's a false negative rather than false positive — i.e. as the code is written now, it will not accidentally merge something that does not share a common part, but it may not merge more complex stuff that does have a common part. I'm a bit afraid that if I try to cover too many corner cases, I'll end up with false positives.

Comment (beckermr) [reply conda_forge_tick/migrators/recipe_v1.py:71]: Ok. Let's add some complex cases that could naively be merged to the test suite to ensure the code does not touch them.

**Comment by mgorny** [comment]
I've added some unit tests for edge cases. Are there some other cases you'd like to see?

**Review by beckermr — APPROVED** [review]
Lgtm! Thank you!

**Comment by mgorny** [comment]
Thanks. BTW after your comment, I'm actually thinking of trying to support the simplest case of extra parentheses.
